exercises/pascals-triangle/src/example.c

```diff
@@ -0,0 +1,39 @@
+#include "pascals_triangle.h"
+
+void pt_free(size_t ** mem_block, size_t n)
+{
+   for (size_t i = 0; i < n; i++) {
+      free(mem_block[i]);
+   }
+   free(mem_block);
+}
+
+size_t **pascals_triangle(char *count)
+{
+   if (count == NULL) {
+      return (size_t **) NULL;
+   }
+   size_t **zero = (size_t **) calloc(1, sizeof(size_t *));
+   zero[0] = (size_t *) calloc(1, sizeof(size_t));
+   int rows = atoi(count);
+   if (!strcmp(count, "0")) {
+      return zero;
+   } else if (rows <= 0) {
+      pt_free(zero, 1);
+      return (size_t **) NULL;
+   } else {
+      pt_free(zero, 1);
+   }
+   size_t **result = (size_t **) calloc(rows, sizeof(size_t *));
```

Comment:
It is not recommended to cast the result of `malloc`/`calloc`: http://stackoverflow.com/questions/605845/do-i-cast-the-result-of-malloc